ppa/archive/admin.py

```diff
@@ -12,23 +12,38 @@ class DigitizedWorkAdmin(admin.ModelAdmin):
         'page_count', 'is_public', 'added', 'updated'
     )
     fields = (
-        'source_link', 'title', 'subtitle', 'sort_title', 'enumcron',
-        'author', 'pub_place', 'publisher', 'pub_date', 'page_count',
-        'public_notes', 'notes', 'record_id', 'collections',
-        'status', 'added', 'updated'
+        'source', 'source_id', 'source_url', 'title', 'subtitle',
+        'sort_title', 'enumcron', 'author', 'pub_place', 'publisher',
+        'pub_date', 'page_count', 'public_notes', 'notes', 'record_id',
+        'collections', 'status', 'added', 'updated'
     )
+    # fields that are always read only
     readonly_fields = (
-        'source_link', 'page_count', 'added', 'updated', 'record_id',
+        'added', 'updated'
     )
+    # fields that are read only for HathiTrust records
+    hathi_readonly_fields = (
+        'source', 'source_id', 'source_url', 'page_count', 'record_id',
+    )
+
     search_fields = (
         'source_id', 'title', 'subtitle', 'author', 'enumcron', 'pub_date',
         'publisher', 'public_notes', 'notes', 'record_id'
     )
     filter_horizontal = ('collections',)
     # date_hierarchy = 'added'  # is this useful?
-    list_filter = ['collections', 'status']
+    list_filter = ['collections', 'status', 'source']
     actions = ['bulk_add_collection']
 
+    def get_readonly_fields(self, request, obj=None):
```

Comment:
The logic behind this is such that if a work is accidentally made HATHI, those fields are no longer editable -- including the `source` field. At which point, the only way around this would be to delete it.

I confirmed this by trying it out, and that makes sense to me, but I want to make sure that's the intended functionality.